01/lib/raii_action.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <functional>
+#include <iostream>
+#include <concepts>
+
+
+template <typename T>
+concept callable =
+    requires(T func)
+    {
+        func();
+    };
+
+template <callable T>
+class RAIIAction {
+private:
+    T func;
+
+public:
+    RAIIAction(T _func) : func(_func) {}
```

Comment:
Имена начинающиеся с `_` зарезервированы для стандартной библиотеки. Можно сделать так
```suggestion
    RAIIAction(T  f) : func_(f) {}
```